app/controllers/CheckDetailsController.scala

```diff
@@ -125,79 +124,93 @@ class CheckDetailsController @Inject()(
-    auditService.findYourNinoPDVMatched(pdvData, origin, idData)
-
-    individualDetailsService.createIndividualDetailsData(sessionWithNINO.data.getOrElse("sessionId", EmptyString), idData)
+    auditService.findYourNinoPDVMatched(pdvData, origin, Some(idData))
 
+    val sessionId = sessionWithNINO.data.getOrElse("sessionId", EmptyString)
     val api1694Checks = checkDetailsService.checkConditions(idData)
-    personalDetailsValidationService.updatePDVDataRowWithValidCustomer(pdvData.getNino, api1694Checks._1, api1694Checks._2)
-    if (api1694Checks._1) {
-      val idPostCode = individualDetailsService.getNPSPostCode(idData)
-      if (pdvData.getPostCode.nonEmpty) {
-        // Matched with PostCode
-        if (comparePostCode(idPostCode, pdvData.getPostCode)) {
-          logger.info(s"PDV and API 1694 postcodes matched")
-          Redirect(routes.ValidDataNINOHelpController.onPageLoad(mode = mode)).withSession(sessionWithNINO)
+
+    val futureResult = individualDetailsService.createIndividualDetailsData(sessionId, idData).flatMap { _ =>
+      personalDetailsValidationService.updatePDVDataRowWithValidCustomer(pdvData.getNino, api1694Checks._1, api1694Checks._2).map { _ =>
+        if (api1694Checks._1) {
+          val idPostCode = individualDetailsService.getNPSPostCode(idData)
+          if (pdvData.getPostCode.nonEmpty) {
+            // Matched with PostCode
+            if (comparePostCode(idPostCode, pdvData.getPostCode)) {
+              logger.info(s"PDV and API 1694 postcodes matched")
+              Redirect(routes.ValidDataNINOHelpController.onPageLoad(mode = mode)).withSession(sessionWithNINO)
+            } else {
+              logger.warn(s"PDV and API 1694 postcodes not matched")
+              redirectToInvalidDataNINOHelpController(mode, sessionWithNINO)
+            }
+          } else { // Matched with NINO
+            personalDetailsValidationService.updatePDVDataRowWithNPSPostCode(pdvData.getNino, idPostCode)
+            Redirect(routes.ValidDataNINOMatchedNINOHelpController.onPageLoad(mode = mode)).withSession(sessionWithNINO)
+          }
         } else {
-          logger.warn(s"PDV and API 1694 postcodes not matched")
-          Redirect(routes.InvalidDataNINOHelpController.onPageLoad(mode = mode)).withSession(sessionWithNINO)
+          logger.warn(s"API 1694 checks failed: ${api1694Checks._2}")
+          redirectToInvalidDataNINOHelpController(mode, sessionWithNINO)
         }
-      } else { // Matched with NINO
-        personalDetailsValidationService.updatePDVDataRowWithNPSPostCode(pdvData.getNino, idPostCode)
-        Redirect(routes.ValidDataNINOMatchedNINOHelpController.onPageLoad(mode = mode)).withSession(sessionWithNINO)
       }
-    } else {
-      logger.warn(s"API 1694 checks failed: ${api1694Checks._2}")
-      Redirect(routes.InvalidDataNINOHelpController.onPageLoad(mode = mode)).withSession(sessionWithNINO)
     }
+
+    Await.result(futureResult, Duration.Inf)
```

Comment:
We should not need to do this. I think I can come up with an alternative when we mob later.  :-)